addons/account/models/account_move.py

```diff
@@ -1994,12 +2019,7 @@ def js_assign_outstanding_line(self, line_id):
         return lines.reconcile()
 
     def button_draft(self):
-        if any(move.state != 'cancel' for move in self):
-            raise UserError(_('Only cancelled journal entries can be reset to draft.'))
-        self.write({'state': 'draft'})
-
-    def button_cancel(self):
-        self._check_tax_lock_date()
+        self._check_move_consistency(check_balanced=False)
```

Comment:
check_balanced=False?

Reply:
The check balanced is done at the end of the write, don't need to perform it twice

Reply:
check that inside the write method (before the "real" write like the 'date')